src/utils/publishToNpm.js

```diff
@@ -63,18 +63,57 @@ async function publishToNpm({
   if (await allowNpmPublish(version)) {
     await execWithOutput('npm', ['pack', '--dry-run'])
 
-    if (opticToken) {
-      const packageInfo = await getLocalInfo()
-      const otp = await execWithOutput('curl', [
-        '-s',
-        '-d',
-        JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
-        '-H',
-        'Content-Type: application/json',
-        '-X',
-        'POST',
-        `${opticUrl}${opticToken}`,
-      ])
+    let otp
+
+    if (opticToken || ngrokApiKey) {
+      if (opticToken) {
+        const packageInfo = await getLocalInfo()
+        
+        // push notification via optic mobile app
+        otp = await execWithOutput('curl', [
+          '-s',
+          '-d',
+          JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
+          '-H',
+          'Content-Type: application/json',
+          '-X',
+          'POST',
+          `${opticUrl}${opticToken}`,
+        ])
+      } else if (ngrokApiKey) {
+        // 1. run a fastify app which exposes a simple web UI with a text input to get the OTP
+        // 2. expose the app via ngrok
+        // 3. print the url to the app on the workflow execution output
+        // 4. wait for some time or a timeout
+        // 5. if we get the token from the user, continue with the execution, otherwise timeout
+        const app = fastify()
+
+        app.get('/', () => {
+          // serve static page with text input which, when submitted, POSTs the OTP to /otp url
```

Comment:
@nickzafiropulos @msgadi can you please work together to create a simple template for this page which is on-brand?

Reply:
@msgadi you could use fastify/static, if you wish, or maybe even keep it simple and just build a simple html page with css and js inline